File: src/main/java/com/warren/bt/Inverter.java
package com.warren.bt;

public class Inverter extends Decorator {
  public Inverter(Node child) {
    super(child);
  }

  public Inverter() {
    super();
  }

  @Override
  public Status tick() {
    if (child == null) {
      return Status.FAILURE;
    }

    Status status = child.tick();

    switch (status) {
      case SUCCESS:
        return Status.FAILURE;
      case FAILURE:
        return Status.SUCCESS;
      default:
        return Status.RUNNING;
    }
  }
}

File: src/main/kotlin/com/warren/bt/Inverter.kt
package com.warren.bt

class Inverter : Decorator("Inverter") {
  override fun tick(): Status {
    val s = child?.tick() ?: return Status.FAILURE
    return when (s) {
      Status.SUCCESS -> Status.FAILURE
      Status.FAILURE -> Status.SUCCESS
      Status.RUNNING -> Status.RUNNING
    }
  }
}
